indico/web/client/js/react/components/PublicationSwitch.jsx

```diff
@@ -0,0 +1,137 @@
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {Translate} from 'indico/react/i18n';
+
+export default function PublicationSwitch({eventId}) {
+  const [published, setPublished] = useState(true);
+  const [modalOpen, setModalOpen] = useState(false);
+
+  useEffect(() => {
+    const fetchPublicationSetting = async () => {
+      let response;
+      try {
+        response = await indicoAxios.get(publicationURL({confId: eventId}));
+      } catch (error) {
+        handleAxiosError(error);
+      }
+      setPublished(response.data.published);
+    };
+    fetchPublicationSetting();
+  }, [eventId]);
+
+  const changeState = shouldPublish => {
+    const updatePublicationSetting = async () => {
+      try {
+        if (shouldPublish) {
+          await indicoAxios.put(publicationURL({confId: eventId}));
+        } else {
+          await indicoAxios.delete(publicationURL({confId: eventId}));
+        }
+      } catch (error) {
+        handleAxiosError(error);
+      }
+      setPublished(shouldPublish);
+    };
+    updatePublicationSetting();
+  };
+
+  const menuList = (
+    <List bulleted>
+      <List.Item>
+        <Translate>Contribution List</Translate>
+      </List.Item>
+      <List.Item>
+        <Translate>My Contributions</Translate>
+      </List.Item>
+      <List.Item>
+        <Translate>Author List</Translate>
+      </List.Item>
+      <List.Item>
+        <Translate>Speaker List</Translate>
+      </List.Item>
+      <List.Item>
+        <Translate>Timetable</Translate>
+      </List.Item>
+      <List.Item>
+        <Translate>Book of Abstracts</Translate>
+      </List.Item>
+    </List>
+  );
+
+  return (
+    <Modal
+      open={modalOpen}
+      onClose={() => setModalOpen(false)}
+      size="tiny"
+      trigger={
+        <Checkbox
+          label={published ? Translate.string('Published') : Translate.string('Draft')}
+          toggle
+          onClick={() => setModalOpen(true)}
+          checked={published}
+        />
+      }
+      closeIcon
+    >
+      <Header
+        content={
+          published
+            ? Translate.string('Set contribution list in Draft mode')
+            : Translate.string('Publish Contributions')
+        }
+      />
+      <Modal.Content>
+        {published ? (
+          <>
+            <p>
+              <Translate>Are you sure you want to set contribution list in draft mode?</Translate>
+            </p>
+            <p>
+              <Translate>By doing so the following menu items won't be accessible:</Translate>
+            </p>
+            {menuList}
+          </>
+        ) : (
+          <>
+            <p>
+              <Translate>Are you sure you want to publish the contribution list?</Translate>
+            </p>
+            <p>
+              <Translate>By doing so the following menu items will be accessible: </Translate>
+            </p>
+            {menuList}
```

Comment:
why not move this after the ternary instead of having it in both places? then you don't even need the variable but can put the JSX there directly